gusto/physics.py

```diff
@@ -0,0 +1,120 @@
+from abc import ABCMeta, abstractmethod
+from firedrake import exp, Function, project, conditional, interpolate
+
+
+class Physics(object):
+    """
+    Base class for physics processes for Gusto.
+
+    :arg state: :class:`.State` object.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, state):
+        self.state = state
+
+    @abstractmethod
+    def apply(self):
+        """
+        Function computes the value of specific
+        fields at the next time step.
+        """
+        pass
+
+
+class Condensation(Physics):
+    """
+    The process of condensation of water vapour
+    into liquid water and evaporation of liquid
+    water into water vapour, with the associated
+    latent heat changes.
+
+    :arg state: :class:`.State.` object.
+    """
+
+    def __init__(self, state):
+        super(Condensation, self).__init__(state)
+
+        # obtain our fields
+        self.theta = getattr(state.fields, 'theta')
```

Comment:
I think the preferred way to get fields when we have the field name, is, for example: 
`self.theta = state.fields("theta")`